tests/trainer/test_trainer.py

```diff
@@ -880,6 +880,46 @@ def training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, hidde
     trainer.fit(model)
 
 
+def test_gradient_clipping_by_value(tmpdir):
+    """
+    Test gradient clipping by value
+    """
+    tutils.reset_seed()
+
+    model = EvalModelTemplate()
+
+    grad_clip_val = 0.0001
+    trainer = Trainer(
+        max_steps=10,
+        max_epochs=1,
+        gradient_clip_val=grad_clip_val,
+        gradient_clip_algorithm='value',
+        default_root_dir=tmpdir,
+    )
+
+    trainer.train_loop.old_training_step_and_backward = trainer.train_loop.training_step_and_backward
+
+    def training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, hiddens):
```

Comment:
can we run this check inside one of the `LightningModule` hooks instead? In case this function ever changes, I'd rather avoid having to update this test

Reply:
https://github.com/dhkim0225/pytorch-lightning/blob/2e933d4aea428b54ed90198b795602bfb3ae7ab3/tests/trainer/test_trainer.py#L847-L880

Right above there is a gradient clip_by_norm test, isn't it weird to put only clip_by_value differently?

Reply:
Didn't see that one. Feel free to update it too

I think it's cleaner to do it in a public hook instead of monkey-patching the loop. It can also serve as documentation by example :)

Reply:
I have read the [hooks.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/hooks.py), but actually, I have no idea how to test it in "a public hook".
Could you please explain a little bit more?
Thanks :]

Reply:
nvm, we don't currently have hooks to easily do this. I'll revisit it in a separate PR.

thanks!